### PR TITLE
React JS client to query elevation service and render colored line segments

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
The latest committed code draws the road elevation each time you draw a polygon and the colored breaks only appear if there are roads present within the polygon.

File: server/src/main/scala/geotrellis/osme/server/OsmeServiceActor.scala [hunk 1]
```diff
@@ -18,25 +18,26 @@ class OsmeServiceActor(
...
+              // we can for and return Features and geometries because we import spray.httpx.SprayJsonSupport._
+              // and import JsonFormat[_] for the required types from geotrellis.vector.io._
+              OsmeService.riverFeatures.filter{ feature =>
+                feature.geom.intersects(queryPolygon)
+              }
             }
           }
         }
       }
     }
```
Comment (echeipesh) [inline server/src/main/scala/geotrellis/osme/server/OsmeServiceActor.scala:40]:
``` sh
[error] /Users/eugene/proj/geotrellis-osm-elevation/server/src/main/scala/geotrellis/osme/server/OsmeServiceActor.scala:39: Unmatched closing brace '}' ignored here
[error]     }
[error]     ^
```

Getting a build error here, looks like an extra brace is present.

File: viewer/package.json [hunk 2]
```diff
@@ -23,13 +23,15 @@
...
     "less": "^2.5.3",
     "less-loader": "^2.2.2",
     "lodash": "^4.0.0",
     "react": "^0.14.3",
     "react-bootstrap": "^0.28.2",
     "react-dom": "^0.14.3",
-    "react-leaflet": "^0.10.0",
+    "react-leaflet": "^0.9.0",
+    "react-leaflet-draw": "^0.5.0",
     "react-redux": "^4.0.0",
```
Comment (echeipesh) [inline viewer/package.json:35]: Getting:

``` console
ERROR in ./utils/renderRoads.js
Module not found: Error: Cannot resolve module 'geocolor' in /Users/eugene/proj/geotrellis-osm-elevation/viewer/utils
 @ ./utils/renderRoads.js 8:15-34
^C
```

I had to install the module manually

```
npm install --save geocolor
```

This change should be reflected here as well.

File: server/src/main/scala/geotrellis/osme/server/OsmeServiceActor.scala (hunk 1 above)
Comment (echeipesh) [inline server/src/main/scala/geotrellis/osme/server/OsmeServiceActor.scala:40]: Actually it seems like it's missing the wrapping `cors {` at the top

**Comment by echeipesh** [comment]
This looks good. If you could correct the two things that prevent this from being built, I'd like to merge this today.

File: viewer/package.json (hunk 2 above)
Comment (ghost) [inline viewer/package.json:35]: For this, do I git push the module or just add the gecolor version to package.json?

**Comment by echeipesh** [comment]
Looks good
osm-legend

**Comment by ghost** [comment]
I may make some changes to the rendering of the legend during the holidays and also implement the deletion of lines along with the polygon. As of now, I simply do not have the time and will have to leave the country soon.

**Pull request merged**